.github/workflows/main.yml

```diff
@@ -31,9 +31,27 @@ jobs:
           node-version: '12.x' # TODO: match nvmrc version
       - name: Install dependencies
         run: yarn --frozen-lockfile
+      - name: Setup sonar scanner
+        uses: warchant/setup-sonar-scanner@v1
       - name: Lint
         run: yarn lint
       - name: Run tests
         run: |
           echo TYPEORM_PASSWORD=${{ secrets.TEST_POSTGRES_PASSWORD }} >> .env.ci
           yarn test:ci
+      # Another existing action, this one to checkout the repository
+      - name: "Checkout repository on branch: ${{ github.REF }}"
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.REF }}
+          fetch-depth: 0
+      # Before running the scan, typescript must be installed from npm and
+      # NODE_PATH env var must point to tsc bin folder, otherways typescript static analysis will fail
+      - name: "Run Sonarqube analysis of the branch"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          yarn install 
+          yarn run build
```

Comment:
Maybe we can move this action to a named step like `Build` and run it right after installing dependencies (line 32) so that if the code does not even build then we don't even run the following checks, WDYT?

Reply:
I had some issues before with this one, the problem here is that the plugin we're using to analyze typescript code requires those steps before running the scan (otherways fails running), but we can test the alternative you mentioned.